kombu/transport/filesystem.py

```diff
@@ -140,88 +144,118 @@ def lock(file, flags):
     def unlock(file):
         """Remove file lock."""
         fcntl.flock(file.fileno(), fcntl.LOCK_UN)
+
+
 else:
-    raise RuntimeError(
-        'Filesystem plugin only defined for NT and POSIX platforms')
+    raise RuntimeError("Filesystem plugin only defined for NT and POSIX platforms")
+
+
+exchange_queue_t = namedtuple("exchange_queue_t", ["routing_key", "pattern", "queue"]) 
 
 
 class Channel(virtual.Channel):
     """Filesystem Channel."""
 
+    supports_fanout = True
+
+    @contextmanager
+    def _get_exchange_file_obj(self, exchange, mode="rb"):
+        filename = f"{exchange}.exchange"
+        filename = os.path.join(self.control_folder, filename)
+        try:
+            f_obj = open(filename, mode)
+            if "w" in mode:
+                lock(f_obj, LOCK_EX)
+            yield f_obj
+        except OSError:
+            raise ChannelError(f"Cannot open {filename}")
+        finally:
+            if "w" in mode:
+                unlock(f_obj)
+            f_obj.close()
+
+    def get_table(self, exchange):
+        try:
+            with self._get_exchange_file_obj(exchange) as f_obj:
+                exchange_table = loads(bytes_to_str(f_obj.read()))
+                return [exchange_queue_t(*q) for q in exchange_table]
```

Comment:
Please use namedtuple here. Define it as global data structure.

from  https://github.com/celery/kombu/pull/1280#discussion_r546299433